### PR TITLE
aegisub: 3.3.3 -> 3.4.0

### PR DESCRIPTION
**Pull request by emilazy**
Aegisub is a program of many forks. We currently use the inactive wangqr fork that has not received an update in two years. This switches to the actively‐maintained arch1t3cht fork, bringing in additional functionality, bug fixes, and a much more reasonable Meson build system that lets us drop our LuaJIT devendoring patch. It also contains fixes for newer versions of Boost.

Note that although the `README.md` has a note from two years ago saying “**Don't** use this version if you're just looking for any version of Aegisub”, that is somewhat outdated and predates it becoming a fully‐fledged fork: as the next section of the file from two months ago admits, it is the current de facto standard Aegisub fork and the version universally recommended by up‐to‐date sources; see e.g.:

* <https://fansubbers.miraheze.org/wiki/Aegisub>
* <https://guide.encode.moe/typesetting/aegisub.html>
* <https://www.reddit.com/r/aegisub/comments/yzvi4m/aegisub_update_status/>
* <https://github.com/Ristellise/AegisubDC/issues/10>

cc @al3xtjames – I see you’ve worked on packaging this for Nix and [contributed upstream](https://github.com/arch1t3cht/Aegisub/pull/162) to assist with that; perhaps you’d be interested in maintaining this package? :) (I expect your version may have benefits over this, from a quick skim; I just did this in the process of Boost clean‐up, but I’d be happy to review a PR bringing it in line with your changes.)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ae/aegisub/package.nix [hunk 1]
```diff
@@ -167,5 +144,7 @@ stdenv.mkDerivation (finalAttrs: {
     mainProgram = "aegisub";
     maintainers = with lib.maintainers; [ wegank ];
     platforms = lib.platforms.unix;
+    # No native wxWidgets and VapourSynth is broken.
```
Comment (snaakey) [inline pkgs/by-name/ae/aegisub/package.nix:147]: I'm unable to test VapourSynth on darwin but if its broken maybe also add `badPlatforms` to the VapourSynth package.

Comment (snaakey) [reply pkgs/by-name/ae/aegisub/package.nix:147]: Oh nvm I should have looked at the whole file before commenting

**Review by snaakey — APPROVED** [review]

**Comment by emilazy** [comment]
Sadly they compile code specific to the wxWidgets Cocoa backend on macOS so even after @wegank’s work to get VapourSynth working on Darwin this still won’t go. Would be nice to get that backend packaged at some point.

**Comment by emilazy** [comment]
> I think [wangqr/Aegisub@3556c96](https://github.com/wangqr/Aegisub/commit/3556c9629ea6adc27a382ab507d5d076131919d2) would help fix aegisub on darwin.

Aha, a regression, interesting! I guess I should port that to the active fork then :)

> (Also, I think I've mentioned this before, but `wxGTK32` is a misnomer on Darwin, as it doesn't depend on GTK at all. Renaming it to `wxwidgets_3_2` would probably be a good idea...)

Oh, I had no idea and just assumed. That is indeed extremely confusing.

**Comment by emilazy** [comment]
Looks like it was intentionally undone: https://github.com/arch1t3cht/Aegisub/commit/6dda04da9fd9b43a22266fdb1c4fa293b418e3db.

I wonder if there’s anything we can do to get at the private symbols here?

**Comment by al3xtjames** [comment]
IIRC [that commit was reverted in the fork to fix IME input](https://github.com/arch1t3cht/Aegisub/issues/91). Unfortunately [it uses private symbols from wxWidgets)[https://github.com/arch1t3cht/Aegisub/issues/154] which causes linking to fail when using a non-bundled version. I worked around this [by using the bundled wx on Darwin](https://gist.github.com/al3xtjames/27c87c36cbbd2245eb46213bf5b9f563#file-aegisub-nix-L59), though this isn't ideal...

**Comment by kini** [comment]
Looks like upstream is no longer dead, and I guess is reabsorbing arch1t3cht's fork? https://aegisub.org/blog/aegisub-3.4.0-released

**Comment by wegank** [comment]
@ofborg build aegisub aegisub.passthru.tests

**Comment by wegank** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 359027`

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 3 packages built:</summary>
  <ul>
    <li>aegisub</li>
    <li>vapoursynth-bestsource</li>
    <li>vapoursynth-bestsource.dev</li>
  </ul>
</details>

**Comment by emilazy** [comment]
Hmm, I feel a little strange about what happened to this PR while I was gone.

The new upstream is the TypesettingTools fork, which I already considered while making this PR. It’s definitely what we should be using in the long run, and I’m glad that the Aegisub ecosystem is finally converging on one repository. However, as the blog post itself says, it does not include all the changes from arch1t3cht’s fork yet:

> The goal was to just cut a release that builds on modern systems, which by itself was more than enough work. Further releases will include actual features and more normal changelogs.

> arch1t3cht has done a great job of running the primary fork used over the past few years, and now has commit access in TypesettingTools so that he can work out of a more centralized repository. We’d love to get all the work done across various forks back into the mainline repository, so please send PRs!

The arch1t3cht fork is still *hundreds* of commits ahead of the TypesettingTools repository, and the only changes to the latter since I opened this PR were some bug fix commits. I think the correct thing to do here is to continue to use arch1t3cht’s fork for now, for the reasons I gave in the original PR message, and then switch to the TypesettingTools repository in a release or two once it has absorbed a meaningful amount of the functionality in the fork. Mostly though it just feels a little odd for my PR with the explicit intent of switching to the arch1t3cht fork for now to get reused for this – I wouldn’t mind if this one had been closed in favour of a PR taking the alternate approach.

**Comment by kini** [comment]
I completely agree. I hope I'm wrong, but if by any chance my comment above was what prompted the repurposing of this PR from "aegisub: switch to arch1t3cht's fork" to "3.3.3 -> 3.4.0" and its subsequent merge, then I apologize as that was not my intention ― I just wanted to bring attention to the blog post announcing the release of 3.4.0 since I thought it relevant to the discussion.

I don't know if my $0.02 is worth anything since I'm basically just a driveby commenter on this PR :sweat_smile: but IMO the author of a PR should be the one to make any major changes to the goal or aim of the PR, and if someone wants to do something different it's easy enough to open a different PR for that.

**Comment by wegank** [comment]
I apologize for the quick and thoughtless bump to 3.4.0. Feel free to revert the latest commit or introduce the fork as a separate package.

**Pull request merged**